Enhancements: Bulk Delete, QR Scanner Integration, Simplified Code Structure, Separated Pages, and Alert on Success

yoooonnn opened the pull request.

yoooonnn committed:
Simplify, Seperate Add, Update Pages #15

yoooonnn committed:
Bulk Remove Function #15

yoooonnn committed:
Complete Alert #15

yoooonnn committed:
delete update function, QR reader #15

dongwook1214 commented on src/App.js:
```diff
@@ -12,7 +13,10 @@ function App() {
         <Route path="/" element={<><Login /><SignUp /></>} />
-        <Route path="/device" element={<><AddDevice /><DeviceList /></>}/>
+        <Route path="/device" element={<DeviceList />}/>
+        <Route path="/addDevice" element={<AddDevice />} />
```
카멜보다는 _ 로 표현하는게 좋을듯!
add_device

dongwook1214 commented on src/App.js:
```diff
@@ -12,7 +13,10 @@ function App() {
-        <Route path="/device" element={<><AddDevice /><DeviceList /></>}/>
+        <Route path="/device" element={<DeviceList />}/>
+        <Route path="/addDevice" element={<AddDevice />} />
+        <Route path="/updateDevice/:publicKey" element={<UpdateDevice />} />
```
update_device

dongwook1214 commented on src/modules/device/states/deviceSlice.js:
```diff
@@ -14,10 +14,28 @@ const deviceSlice = createSlice({
             state.push(action.payload)
```
local storage 에도 반영되면 좋을듯

dongwook1214 commented on src/modules/device/components/AddDevice.js:
```diff
@@ -1,34 +1,94 @@
+    // QR 코드 스캔 후, publicKey에 맞는 deviceType을 자동으로 설정
+    const handleScan = (data) => {
+        if (data) {
+            setPublicKey(data);  // QR 코드에서 받은 publicKey를 설정
```
바로 addDevice 하면 좋겠는데
일단 지수누나한테 맡기자

dongwook1214 commented on src/modules/device/components/AddDevice.js:
```diff
@@ -1,34 +1,94 @@
+                    name="publicKey"
+                    value={publicKey}
+                    placeHolder="디바이스 코드를 입력해주세요."
+                    onChange={onPublicKeyChange}
```
여기 좀 이상한데
직접 입력해보면 입력할 때마다 qr scanner가 깜빡거림

dongwook1214 replied:
아마 hook 쓰면 state 바뀔 때마다 전체 페이지를 다시 그려서 그러거 같은데
memo 써서 QrReader 따로 분리하고
handleScan, handleError 함수도 useCallback 으로 바꾸면 될지도??

yoooonnn committed:
Sync devices list with local storage #17

yoooonnn committed:
Optimize Rendering and Auto Register Devices After QR Scan #17

- Optimization: Minimized unnecessary re-renders by utilizing `useMemo` and `useCallback`
- Feature Improvement: Automatically calls `addDevice` after scanning QR code to register devices

dongwook1214 commented:
LGTM
